plaidml/plaidml++.h

```diff
@@ -169,9 +169,13 @@ class base_shape {
 template <typename T>
 class shape : public base_shape {
  public:
-  explicit shape(const std::shared_ptr<ctx>& ctx, datatype dt = to_plaidml_datatype<T>::value) : base_shape(ctx, dt) {}
+  explicit shape(const std::shared_ptr<ctx>& ctx) : shape(ctx, to_plaidml_datatype<T>::value) {}
 
-  explicit shape(const base_shape& base, datatype dt = to_plaidml_datatype<T>::value) : base_shape(base) {
+  explicit shape(const std::shared_ptr<ctx>& ctx, datatype dt) : base_shape(ctx, dt) {}
```

Comment:
`explicit` is not useful for constructors with multiple arguments.

Reply:
It is useful when you want to disallow user from implicitly using initializer list. say you have a function `void blah(const shape& s) { /*whatever*/ }`. Without `explicit`, `blah({ctx, dt})` will be valid.